### PR TITLE
Cleaned up c- and h-files according to style guide

### PR DESCRIPTION
**Pull request by marcusnaslund**

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Any objections @sebastianbaginski ?

File: source/system/Array.h [hunk 1]
```diff
@@ -1,5 +1,5 @@
-#ifndef OOC_LANG_ARRAY_H
-#define OOC_LANG_ARRAY_H
+#ifndef ARRAY_H_
```
Comment (sebastianbaginski) [inline source/system/Array.h]: I'd leave the longer macro guard here, `ARRAY_H_` seems not unique enough to me.

Comment (marcusnaslund) [inline source/system/Array.h]: Good point.

Comment (marcusnaslund) [inline source/system/Array.h]: Changed it to `OOC_ARRAY_H`, since the lang namespace is gone.

Comment (sebastianbaginski) [inline source/system/Array.h]: Great, thanks.

**Pull request merged**